src/Tribe/Ticket_Object.php

```diff
@@ -507,11 +516,22 @@ public function is_in_stock() {
 				return true;
 			}
 
-			$remaining = $this->inventory();
+			/* @var Tribe__Cache $cache */
+			$cache = tribe( 'cache' );
+			$key   = __METHOD__ . '-' . $this->ID;
+
+			if ( isset( $cache[ $key ] ) ) {
+				return $cache[ $key ];
+			}
+			// Storing the $key.
+			$this->cached_keys[] = $key;
 
-			$is_unlimited = $remaining === -1;
+			$remaining    = $this->inventory();
+			$is_unlimited = $remaining === - 1;
```

Comment:
:no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: c9ddc7cdd4efb78e0b6855de6ec877d4</sub>